### ISSUE TITLE
Incomplete W&B Logging - Missing Critical Training Metrics

### ISSUE THREAD

**Issue by DataCTE** [DataCTE/SDXL-Training-Improvements#4]
### Description
Current W&B logging implementation is missing several critical metrics and visualizations needed for proper training monitoring. Need to implement comprehensive logging of all training aspects.

### Missing Metrics
1. VAE Training:
   - Reconstruction loss
   - Latent space statistics
   - VAE gradients

2. UNet Training:
   - Per-step sigma values
   - Noise prediction accuracy
   - Gradient norms

3. Validation:
   - ZTSNR effectiveness metrics
   - High-resolution coherence scores
   - Sample image quality metrics

### Implementation Plan
```python
# Add to training loop:
if args.use_wandb:
    wandb.log({
        # Training metrics
        'train/unet_loss': loss.item(),
        'train/weighted_loss': weighted_loss.item(),
        'train/grad_norm': grad_norm,
        'train/sigma': sigma.mean().item(),
        
        # VAE metrics
        'vae/reconstruction_loss': vae_loss,
        'vae/latent_mean': latent_mean,
        'vae/latent_std': latent_std,
        
        # Learning rates
        'lr/unet': lr_scheduler.get_last_lr()[0],
        'lr/vae': vae_lr_scheduler.get_last_lr()[0],
        
        # System metrics
        'system/gpu_memory': torch.cuda.memory_allocated(),
        'system/gpu_utilization': gpu_utilization,
    })
```

### Additional Features Needed
1. Custom W&B panels for:
   - Training progress visualization
   - Sample image comparison
   - Validation metrics tracking
   - System resource monitoring

2. Automatic logging of:
   - Model architecture
   - Training configuration
   - System information
   - Git commit information

### Priority: High
Proper logging is crucial for debugging and monitoring training progress.
